sycl/plugins/unified_runtime/pi2ur.hpp

```diff
@@ -1987,8 +1987,19 @@ piProgramLink(pi_context Context, pi_uint32 NumDevices,
   ur_program_handle_t *UrProgram =
       reinterpret_cast<ur_program_handle_t *>(RetProgram);
 
-  HANDLE_ERRORS(urProgramLink(UrContext, NumInputPrograms, UrInputPrograms,
-                              Options, UrProgram));
+  auto UrDevices = reinterpret_cast<ur_device_handle_t *>(
+      const_cast<pi_device *>(DeviceList));
+
+  auto urResult = urProgramLinkExp(UrContext, NumInputPrograms, UrInputPrograms,
+                                   NumDevices, UrDevices, Options, UrProgram);
+  if (urResult == UR_RESULT_ERROR_UNSUPPORTED_FEATURE) {
+    HANDLE_ERRORS(urProgramLink(UrContext, NumInputPrograms, UrInputPrograms,
+                                Options, UrProgram));
+  } else {
+    if (urResult != UR_RESULT_SUCCESS) {
+      return ur2piResult(urResult);
+    }
+  }

```

Comment:
```suggestion
  if (urResult == UR_RESULT_ERROR_UNSUPPORTED_FEATURE) {
    HANDLE_ERRORS(urProgramLink(UrContext, NumInputPrograms, UrInputPrograms,
                                Options, UrProgram));
  }
  return ur2piResult(urResult);
```

this is cleaner, as ur2piResult already handles PI_SUCCESS case as well.